### PR TITLE
style: Handle correctly document state invalidation inside negation. 

### PR DESCRIPTION
**Pull request by emilio**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19805)

### CODE REVIEW

File: components/selectors/matching.rs [hunk 1]
```diff
@@ -620,6 +620,15 @@ where
     E: Element,
     F: FnMut(&E, ElementSelectorFlags),
 {
+    if cfg!(debug_assertions) {
+        if context.shared.nesting_level == 0 {
+            debug_assert!(!context.shared.in_negation, "How?");
```
Comment (upsuper) [inline components/selectors/matching.rs]: It seems to me that the whole `if` block is equivalent to
```rust
debug_assert!(context.shared.nesting_level > 0 || !context.shared.in_negation);
```

I cannot think out a case which this single line wouldn't assert but the code here would, and vice verse

**Review by upsuper — APPROVED** [review]
I don't particularly like this approach... but I cannot think out a better way either, so fine...

r=me with the comment addressed.

File: components/selectors/matching.rs (hunk 1 above)
Comment (emilio) [reply components/selectors/matching.rs]: Oh, nice... Indeed :)

**Comment by emilio** [comment]
@bors-servo r=upsuper

 * trying to unstuck homu

**Pull request merged**